lib/devise-security/models/session_limitable.rb

```diff
@@ -31,12 +31,71 @@ def update_unique_session_id!(unique_session_id)
         end
       end
 
+      # Removes old/inactive session if allowed.
+      def allow_limitable_authentication?
+        # Always fallback to default logic when session_traceable is not supported.
+        return true unless devise_modules.include?(:session_traceable)
+
+        opts = session_traceable_condition(active: true)
+        return true if max_active_sessions > session_traceable_adapter.find_all(opts).size
+        return deactivate_expired_sessions! if reject_sessions?
+
+        opts[:order] = %i[last_accessed_at asc]
+        session = session_traceable_adapter.find_first(opts)
+        return false unless session
+
+        session.update_attribute_without_validatons_or_callbacks(:active, false)
+      end
+
+      # Deactivate expired sessions
+      def deactivate_expired_sessions!
+        # Always fallback to default logic when session_traceable is not supported.
+        return true unless devise_modules.include?(:session_traceable)
+
+        opts = session_traceable_condition(active: true, order: %i[last_accessed_at asc])
+        session_traceable_adapter.find_all(opts).any? do |session|
+          next unless session.last_accessed_at && timeout_in && session.last_accessed_at <= timeout_in.ago
+
+          session.update_attribute_without_validatons_or_callbacks(:active, false)
+        end
+      end
+
+      # Maximum number of active sessions
+      # @return [Numeric]
+      # @return [1] by default. This can be overridden by application logic as necessary.
+      def max_active_sessions
+        # Always fallback to default logic when session_traceable is not supported.
+        return 1 unless devise_modules.include?(:session_traceable)
```

Comment:
Why do we need to have `session_traceable` turned on before allowing more than 1 active session?